python_modules/dagster/dagster/core/log.py

```diff
@@ -80,14 +126,46 @@ def _log(self, method, orig_message, message_props):
         #     message_with_structured_props, extra={DAGSTER_META_KEY: all_props}
         # )
 
-        for logger in self.loggers:
-            logger_method = check.is_callable(getattr(logger, method))
-            if logger.name == DAGSTER_DEFAULT_LOGGER:
-                logger_method(
-                    msg_with_multiline_structured_props, extra={DAGSTER_META_KEY: all_props}
-                )
+        # HACK: getLevelName really doesn't do what it says on the box. This function was originally
+        # intended to give the user-facing name of a given int log level, e.g.,
+        # getLevelName(40) -> 'ERROR'. In fact, it also works the other way around (this was 'fixed'
+        # in Python 3.4 then immediately reverted in 3.4.2). For levels it doesn't recognize,
+        # including all ints that don't have a name registered with logging.addLevelName it does:
+        # getLevelName('foo') -> 'Level foo'; getLevelName(3) -> 'Level 3' (!)
+        # See: https://docs.python.org/3/library/logging.html#logging.getLevelName
+        if isinstance(level, int):
+            lvl = level
+        else:
+            lvl = logging.getLevelName(level.upper())
+
+        if not isinstance(lvl, int):
+            self.error(
```

Comment:
i go back and forth about whether or not this should be a hard error

Reply:
I think hard error but I have a hard time articulating why

Reply:
In the status quo, if we call `context.log.garbage`, this is a hard error at runtime

Reply:
I've made this a hard error in bdaf5d0